cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java

```diff
@@ -344,14 +329,15 @@ private void renderMsgAndSaveConsent() throws ConsentLibException {
             @Override
             public void onSuccess(Object result) {
                 try{
-                    JSONObject jsonResult = (JSONObject) result;
+                    JSONObject jsonResult = new JSONObject((String) result);
```

Comment:
just curiosity, what's the difference between `(JSONObject) result` vs `new JSONObject((String) result)`. Were we getting casting exception?

Reply:
yes, were were getting casting exception. The JSONObject constructor explicitly parse the the string to a JSONObject while the dynamic cast does not, It simply try to match the casted class as a parent and in this case it breaks as those classes are not living under the same roof 😅